compiler/rustc_ast_lowering/src/lib.rs

```diff
@@ -95,6 +96,13 @@ struct LoweringContext<'a, 'hir> {
     /// Used to allocate HIR nodes.
     arena: &'hir hir::Arena<'hir>,
 
+    /// Used to allocate temporary AST nodes for use during lowering.
+    /// This allows us to create "fake" AST -- these nodes can sometimes
+    /// be allocated on the stack, but other times we need them to live longer
+    /// than the current stack frame, so they can be collected into vectors
+    /// and things like that.
+    lowering_arena: &'a LoweringArena,
```

Comment:
Naming: everything is related to lowering here. `ast_arena`?

Reply:
Done